### PR TITLE
[TTAHUB-2999] Create monthly-delivery-report.sql

### PR DESCRIPTION
**Pull request by GarrettEHill**
## Description of change
JDI query to provide the ability to run a monthly delivery report

## How to test
Read comment at top of file, run the two set configs and the query in the same transaction.

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-2999

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

**Review by thewatermethod — APPROVED** [review]

File: src/queries/monthly-delivery-report.sql [hunk 1]
```diff
@@ -0,0 +1,246 @@
+/**
+* This query collects all the Monitoring goals used on approved reports within the defined time range.
+*
+* The query results are filterable by the JDI flags. All JDI flags are passed as an array of values
```
Comment (AdamAdHocTeam) [inline src/queries/monthly-delivery-report.sql:4]: JDI flags 🤣

File: src/queries/monthly-delivery-report.sql [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+      a.id,
+      a."userId",
+      a.duration,
+      a."deliveryMethod",
+      a."calculatedStatus",
+      a."startDate"
+    FROM "ActivityReports" a
+    WHERE a."calculatedStatus" = 'approved'
+    -- Filter for regionIds if jdi.regionIds is defined
+    AND (NULLIF(current_setting('jdi.regionIds', true), '') IS NULL
```
Comment (AdamAdHocTeam) [inline src/queries/monthly-delivery-report.sql:26]: Not sure if its possible to just have a base filtered table or activity reports using the flags once?

Comment (GarrettEHill) [reply src/queries/monthly-delivery-report.sql:26]: I don't understand what you are asking for.

**Comment by kryswisnaskas** [comment]
Garrett has graciously agreed to change the name from JDI to something more neutral: SSDI - Self Serve Data Interface. This change accommodates those of us who aren't entirely comfortable with the current name and avoids tying the interface to a particular person. What does everyone think about this new proposed name?

**Pull request merged**